### PR TITLE
Fix a type error when trying to set properties for AMQP 1.0

### LINKED ISSUE #50: Error when trying to set AMQP 1.0 properties via rabbitmqctl or HTTP API.
When I try to set AMQP 1.0 message properties with HTTP API or rabbitmqctl there is an internal error. Properties are required for some brokers to route messages.

Example:

```
./sbin/rabbitmqctl set_parameter shovel to_ibm '{"dest-address":"localhost","dest-properties":{"to": "http://localhost:5672/localhost"},"dest-protocol":"amqp10","dest-uri":"amqp://localhost:5672?sasl=anon","src-protocol":"amqp091","src-queue":"foo","src-uri":"amqp://localhost:5673"}'

Setting runtime parameter "shovel" for component "to_ibm" to "{"dest-address":"localhost","dest-properties":{"to": "http://localhost:5672/localhost"},"dest-protocol":"amqp10","dest-uri":"amqp://localhost:5672?sasl=anon","src-protocol":"amqp091","src-queue":"foo","src-uri":"amqp://localhost:5673"}" in vhost "/" ...


Error:
{{:case_clause, {:error, {{:shutdown, {:failed_to_start_child, {"/", "to_ibm"}, {:function_clause, [{:lists, :map, [#Function<9.118865745/1 in :rabbit_shovel_parameters>, %{"to" => "http://localhost:5672/localhost"}], [file: 'lists.erl', line: 1238]}, {:rabbit_shovel_parameters, :parse_amqp10_dest, 4, [file: 'src/rabbit_shovel_parameters.erl', line: 283]}, {:rabbit_shovel_parameters, :parse, 3, [file: 'src/rabbit_shovel_parameters.erl', line: 251]}, {:rabbit_shovel_worker, :init, 1, [file: 'src/rabbit_shovel_worker.erl', line: 53]}, {:gen_server2, :init_it, 6, [file: 'src/gen_server2.erl', line: 548]}, {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 249]}]}}}, {:child, :undefined, {"/", "to_ibm"}, {:rabbit_shovel_dyn_worker_sup, :start_link, [{"/", "to_ibm"}, [{"dest-address", "localhost"}, {"dest-properties", %{"to" => "http://localhost:5672/localhost"}}, {"dest-protocol", "amqp10"}, {"dest-uri", "amqp://localhost:5672?sasl=anon"}, {"src-protocol", "amqp091"}, {"src-queue", "foo"}, {"src-uri", "amqp://localhost:5673"}]]}, :transient, 30000, :worker, [:rabbit_shovel_dyn_worker_sup]}}}}, [{:rabbit_shovel_dyn_worker_sup_sup, :start_child, 2, [file: 'src/rabbit_shovel_dyn_worker_sup_sup.erl', line: 45]}, {:rabbit_runtime_parameters, :set_any0, 5, [file: 'src/rabbit_runtime_parameters.erl', line: 164]}, {:rabbit_runtime_parameters, :set_any, 5, [file: 'src/rabbit_runtime_parameters.erl', line: 143]}, {:rpc, :"-handle_call_call/6-fun-0-", 5, [file: 'rpc.erl', line: 197]}]}

```

### PR DESCRIPTION
**Pull request by hairyhum**
Fixes #50.
[#163887139]

### CODE REVIEW

File: src/rabbit_shovel_parameters.erl [hunk 1]
```diff
@@ -271,7 +271,10 @@ parse_dest(VHostName, ClusterName, Def, SourceHeaders) ->
 parse_amqp10_dest({_VHost, _Name}, _ClusterName, Def, SourceHeaders) ->
     Uris = get_uris(<<"dest-uri">>, Def),
     Address = pget(<<"dest-address">>, Def),
-    Properties = pget(<<"dest-properties">>, Def, []),
+    Properties = case pget(<<"dest-properties">>, Def, []) of
```
Comment (lukebakken) [inline src/rabbit_shovel_parameters.erl]: This came up several times in the management plugin, so @michaelklishin added `rabbit_data_coercion:to_proplist`. I think you can use that here.

Comment (michaelklishin) [inline src/rabbit_shovel_parameters.erl]: You can use `rabbit_data_coercion` functions here.

**Comment by hairyhum** [comment]
Updated with `rabbit_data_coercion:to_proplist` and addressed more fields.

**Comment by hairyhum** [comment]
Added tests. More tests in the managemnet plugin https://github.com/rabbitmq/rabbitmq-shovel-management/pull/32

**Comment by michaelklishin** [comment]
Backported to `v3.7.x`.

**Pull request merged**